packages/mui-base/src/useSelect/useSelect.test.tsx

```diff
@@ -20,4 +21,91 @@ describe('useSelect', () => {
       expect(result.current.getOptionMetadata('c')?.disabled).to.equal(true);
     });
   });
+
+  describe('getHiddenInputProps', () => {
+    it('returns props for hidden input', () => {
+      const options = [
+        { value: 'a', label: 'A' },
+        { value: 'b', label: 'B' },
+        { value: 'c', label: 'C', disabled: true },
+      ];
+
+      const { result } = renderHook(() =>
+        useSelect({ options, defaultValue: 'b', name: 'foo', required: true }),
+      );
+
+      sinon.assert.match(result.current.getHiddenInputProps(), {
+        name: 'foo',
+        tabIndex: -1,
+        'aria-hidden': true,
+        required: true,
+        value: 'b',
+        style: {
+          clip: 'rect(1px, 1px, 1px, 1px)',
+          clipPath: 'inset(50%)',
+          height: '1px',
+          width: '1px',
+          margin: '-1px',
+          overflow: 'hidden',
+          padding: 0,
+          position: 'absolute',
+          left: '50%',
+          bottom: 0,
+        },
+      });
+    });
+
+    it('[multiple] returns correct value for the hidden input', () => {
+      const options = [
+        { value: 'a', label: 'A' },
+        { value: 'b', label: 'B' },
+        { value: 'c', label: 'C', disabled: true },
+      ];
+
+      const { result } = renderHook(() =>
+        useSelect({
+          multiple: true,
+          options,
+          defaultValue: ['a', 'b'],
+          name: 'foo',
+          required: true,
+        }),
+      );
+
+      sinon.assert.match(result.current.getHiddenInputProps(), {
+        name: 'foo',
+        tabIndex: -1,
+        'aria-hidden': true,
+        required: true,
+        value: JSON.stringify(['a', 'b']),
```

Comment:
@michaldudak without removing the logic in https://github.com/mui/material-ui/pull/38167/files#r1298380325, the value will be `['a', 'b']` instead of `'["a","b"]'`.